### PR TITLE
Fix static 404

### PR DESCRIPTION
**Pull request by dpordomingo**
Nonexistent files under `/static` folder will be served as `404` instead of serving `/index.html`

### CODE REVIEW

File: server/handler/static_files.go [hunk 1]
```diff
@@ -33,19 +35,25 @@ type options struct {
 
 // ServeHTTP serves any static file from static directory or fallbacks on index.hml
 func (s *Static) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	filepath := s.dir + r.URL.Path
+	filepath := path.Join(s.dir, r.URL.Path)
 	b, err := assets.Asset(filepath)
 	if err != nil {
+		if strings.HasPrefix(filepath, path.Join(s.dir, "static")) {
```
Comment (smacker) [inline server/handler/static_files.go]: maybe it's better to avoid hard-coded directory name here and put in constant?

**Comment by smacker** [comment]
👍 for the change. But code coverage is down not sure if it's a problem.

File: server/handler/static_files.go (hunk 1 above)
Comment (dpordomingo) [reply server/handler/static_files.go]: This file contains other two strings that should be extracted to constants:
- `window.REPLACE_BY_SERVER`
- `index.html`
For consistency, I'll define all the three constants if you agree @smacker

**Comment by dpordomingo** [comment]
About codecov I created a separated issue for that https://github.com/src-d/gitbase-playground/issues/32
imo, since we've not proper tests here, we should not focus in that in this PR

File: server/handler/static_files.go (hunk 1 above)
Comment (smacker) [reply server/handler/static_files.go]: that would be great, thanks!

Comment (dpordomingo) [reply server/handler/static_files.go]: done in the rebased/amended commit, thanks

**Comment by dpordomingo** [comment]
@carlosms , as described in the PR desc, nonexistent files under `/static` are being served as `index.html` what is wrong; they should be served as `404` instead

**Comment by carlosms** [comment]
Sounds good. I was just wondering if our code was requesting non-existing files.

**Pull request merged**